packages/cozy-konnector-libs/src/libs/linkBankOperations.js

```diff
@@ -42,14 +42,19 @@ class Linker {
   }
 
   async removeBillsFromOperations(bills, operations) {
-    const billsOperations = operations.filter(op => op.bills)
-    const ids = bills.map(bill => `io.cozy.bills:${bill._id}`)
     await Promise.all(
-      billsOperations.map(async op => {
-        op.bills = op.bills || []
-        const bills = op.bills.filter(billId => !ids.includes(billId))
-        if (op.bills.length > bills.length) {
-          await this.updateAttributes(DOCTYPE_OPERATIONS, op, { bills })
+      bills.map(async bill => {
+        for (let op of operations) {
+          const billLongId = `io.cozy.bills:${bill._id}`
+          op.bills = op.bills || []
+          // if bill id found in op bills, do something
+          if (op.bills.indexOf(billLongId) >= 0) {
+            let billsAttribut = op.bills.filter(billId => (billId != billLongId))
+            if (bill.original) {
+              billsAttribut.push(`io.cozy.bills:${bill.original}`)
```

Comment:
With this alg, bill.original may end twice in the bill (in most cases actually). You should also filter out bill.original